clash-prelude/src/Clash/Tutorial.hs

```diff
@@ -1270,6 +1271,8 @@ a general listing of the available template holes:
 * @~ISLIT[N]@: Is the @(N+1)@'th argument to the function a literal.
 * @~ISVAR[N]@: Is the @(N+1)@'th argument to the function explicitly not a
   literal.
+* @~ISSCALAR[N]@: Is the @(N+1)@'th argument to the function a 1-bit scalar. Note
+  that this means VHDL types like `integers`, `real` are not considered scalar.
```

Comment:
This is fine, but doesn't mention as much detail as I wanted. I think a note in `clash-lib/src/Clash/Netlist/BlackBox/Util.hs` like this suffices:

```haskell
{- NOTE [scalar netlist types]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In VHDL and (System)Verilog, there are some differences between the types
considered to be scalar. In Verilog, the `integer`, `real` and `time` types
can be indexed into, whereas in VHDL they are scalar types. This means when
writing Clash blackboxes, ~ISSCALAR may not be consistent with the targeted
HDLs notion of a scalar type.

Currently, Clash treats only `Bit` and `Bool` as scalar. This means it may
result in incorrect VHDL if ~ISSCALAR is used to determine whether a indexing
is used. If we extend this to include types VHDL considers scalar, then we
must also ensure that Clash internally also treats these types as scalar to        
avoid confusion.
-}
```

and then a reference to the note where the check is implemented:

```haskell
(IsScalar n) -> let (_,ty,_) = bbInputs b !! n
                      in case ty of
                        -- See NOTE [scalar netlist types]
                        Bit  -> 1
                        Bool -> 1
                        _    -> 0
```